snap/snapcraft.yaml

```diff
@@ -138,7 +138,7 @@ parts:
       # Ensure we don't have a dubious ownership error from git with a remote build.
       git config --global --add safe.directory $CRAFT_PART_SRC
       # set the version
-      version="$(python3 setup.py --version)"
```

Comment:
This change seems to be always making the charmcraft snap that gets built report the version `dev`, I'm guessing because it's not recognised as installed at this point, so there's not a `_version.py` file in the charmcraft directory from which to report.

Reply:
Doesn't setuptools-scm take care of generating the `_version.py` file, transparently? Have you seen any issues in the build process? :astonished:

Reply:
The [snap artefact from the build](https://github.com/canonical/charmcraft/actions/runs/8500696494?pr=1580) is reporting that, and I've been unable to come up with something else that makes it work as intended. Could you revert the removal of `setup.py` for now, since having it there shouldn't break things?

Reply:
`setup.py` restored. The built `snap.zip` bears a version number again.